compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -258,6 +274,12 @@ static std::tuple<Type, Type, Type> getABCElementTypes(MLIRContext *context,
   case MMAIntrinsic::WMMA_F16_16x16x16_F16: {
     return {f16, f16, f16};
   }
+  case MMAIntrinsic::WMMA_F32_16x16x16_BF16: {
+    return {bf16, bf16, f32};
+  }
+  case MMAIntrinsic::WMMA_BF16_16x16x16_BF16: {
+    return {bf16, bf16, bf16};
+  }
```

Comment:
gcc build error somewhere in this function: https://github.com/iree-org/iree/actions/runs/11814370974/job/32913293915#step:4:7077

```
[6444/8163] Building CXX object compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/CMakeFiles/iree_compiler_Codegen_Dialect_GPU_IR_IREEGPUDialect.objects.dir/IREEGPUAttrs.cpp.o
FAILED: compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/CMakeFiles/iree_compiler_Codegen_Dialect_GPU_IR_IREEGPUDialect.objects.dir/IREEGPUAttrs.cpp.o 
/usr/local/bin/ccache /usr/bin/g++-9  -I/__w/iree/iree -I/__w/iree/iree/build-gcc -I/__w/iree/iree/third_party/llvm-project/llvm/include -I/__w/iree/iree/build-gcc/llvm-project/include -I/__w/iree/iree/third_party/llvm-project/mlir/include -I/__w/iree/iree/build-gcc/llvm-project/tools/mlir/include -I/__w/iree/iree/third_party/llvm-project/lld/include -I/__w/iree/iree/build-gcc/llvm-project/tools/lld/include -I/__w/iree/iree/compiler/src -I/__w/iree/iree/build-gcc/compiler/src -I/__w/iree/iree/runtime/src -I/__w/iree/iree/build-gcc/runtime/src -I/__w/iree/iree/build-gcc/runtime/src/iree/base/internal/flatcc -I/__w/iree/iree/compiler/bindings/c/. -I/__w/iree/iree/build-gcc/compiler/bindings/c/. -isystem /__w/iree/iree/third_party/flatcc/include -O3  -fPIC -fvisibility=hidden -fno-rtti -fno-exceptions -Wall -Werror -Wno-error=deprecated-declarations -Wno-address -Wno-address-of-packed-member -Wno-comment -Wno-format-zero-length -Wno-uninitialized -Wno-overloaded-virtual -Wno-invalid-offsetof -Wno-sign-compare -Wno-unused-function -Wno-unknown-pragmas -Wno-unused-but-set-variable -Wno-misleading-indentation -fmacro-prefix-map=/__w/iree/iree=iree -I/__w/iree/iree/third_party/flatcc/include/ -I/__w/iree/iree/third_party/flatcc/include/flatcc/reflection/ -std=gnu++17 -MD -MT compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/CMakeFiles/iree_compiler_Codegen_Dialect_GPU_IR_IREEGPUDialect.objects.dir/IREEGPUAttrs.cpp.o -MF compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/CMakeFiles/iree_compiler_Codegen_Dialect_GPU_IR_IREEGPUDialect.objects.dir/IREEGPUAttrs.cpp.o.d -o compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/CMakeFiles/iree_compiler_Codegen_Dialect_GPU_IR_IREEGPUDialect.objects.dir/IREEGPUAttrs.cpp.o -c /__w/iree/iree/compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp
/__w/iree/iree/compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp: In function 'std::tuple<mlir::Type, mlir::Type, mlir::Type> mlir::iree_compiler::IREE::GPU::getABCElementTypes(mlir::MLIRContext*, mlir::iree_compiler::IREE::GPU::MMAIntrinsic)':
/__w/iree/iree/compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp:293:1: error: control reaches end of non-void function [-Werror=return-type]
  293 | }
      | ^
cc1plus: all warnings being treated as errors
```

unhandled switch case?

Reply:
Thanks for reporting! I'll fix asap (just bad combination right now of: I'm WFH, the server i've been dev on is down, and my workstation in the office is turned off).

Reply:
And this is not a missing switch case (Clang would complain in that case) but GCC rightfully pointing out that even though we have handled all switch cases, it is still unsafe to assume that one of them is taken. See https://github.com/llvm/llvm-project/issues/115345.

Reply:
https://github.com/iree-org/iree/pull/19140

Reply:
Triggered a run to check the build: https://github.com/iree-org/iree/actions/runs/11826327241/job/32952060616 (should complete in 3-4 hours)